presto-spi/src/main/java/com/facebook/presto/spi/block/RunLengthEncodedBlock.java

```diff
@@ -375,4 +375,21 @@ public boolean isNullUnchecked(int internalPosition)
         assert internalPositionInRange(internalPosition, 0, getPositionCount());
         return value.isNull(0);
     }
+
+    @Override
+    public Block appendNull()
+    {
+        int[] ids = new int[positionCount + 1];
+
+        Block dictionary = value;
+        if (value.isNull(0)) {
```

Comment:
perhaps, make an RLE block here; no need to wrap in Dictionary

Reply:
Made it to the following:
```
@Override
    public Block appendNull()
    {
        if (value.isNull(0)) {
            return new RunLengthEncodedBlock(value, positionCount + 1);
        }
        else {
            Block dictionary = value.appendNull();
            int[] ids = new int[positionCount + 1];
            ids[positionCount] = 1;
            return new DictionaryBlock(dictionary, ids);
        }
    }
```